plugin/kubernetes/controller.go

```diff
@@ -510,6 +519,74 @@ func endpointsEquivalent(a, b *object.Endpoints) bool {
 	return true
 }
 
+func servicesModified(oldSvc, newSvc *object.Service) bool {
+	if oldSvc == nil || newSvc == nil {
+		return true
+	}
+
+	// ExternalName is mutable
+	if oldSvc.ExternalName != newSvc.ExternalName {
+		return true
+	}
+
+	// All Port fields are mutable
+	if len(oldSvc.Ports) != len(newSvc.Ports) {
+		return true
+	}
+	for i := range oldSvc.Ports {
+		if oldSvc.Ports[i].Name != newSvc.Ports[i].Name {
+			return true
+		}
+		if oldSvc.Ports[i].Port != newSvc.Ports[i].Port {
+			return true
+		}
+		if oldSvc.Ports[i].Protocol != newSvc.Ports[i].Protocol {
+			return true
+		}
+	}
+
+	return false
+}
+
+func extServicesModified(oldSvc, newSvc *object.Service) bool {
+	if oldSvc == nil || newSvc == nil {
+		// deleted service is only modified if it had external ips
+		if oldSvc != nil && len(oldSvc.ExternalIPs) > 0 {
+			return true
+		}
+		// added service is only modified if it has external ips
+		if newSvc != nil && len(newSvc.ExternalIPs) > 0 {
+			return true
+		}
+		return false
+	}
+
+	// External IPs are mutable
+	if len(oldSvc.ExternalIPs) != len(newSvc.ExternalIPs) {
+		return true
+	}
+	for i := range oldSvc.ExternalIPs {
+		if oldSvc.ExternalIPs[i] != newSvc.ExternalIPs[i] {
+			return true
+		}
+	}
+
+	// All Port fields are mutable
+	for i := range oldSvc.Ports {
```

Comment:
you forgot to check the len on the ports this time, so if newPorts > oldPorts this can return a false negative.

Reply:
oops. good catch.